source/common/http/conn_manager_impl.cc

```diff
@@ -886,15 +884,15 @@ void ConnectionManagerImpl::ActiveStream::decodeHeaders(RequestHeaderMapPtr&& he
   // have broken the path into pieces if applicable. NOTE: Currently the HTTP/1.1 codec only does
   // this when the allow_absolute_url flag is enabled on the HCM.
   if ((!HeaderUtility::isConnect(*request_headers_) && !request_headers_->Path()) ||
```

Comment:
I think this one can be simplified to just check for !connect && empty?

Reply:
we could, but then we'd allow connect with explicitly specified but empty path, which feels buggy to me when everywhere else it needs to have /
WDYT?

Reply:
OK that makes sense, but tbh I've spent too much time staring at this. Can you maybe add some more comments and/or change the conditionals to be more clear somehow?